weldx/asdf/schemas/weldx.bam.de/weldx/debug/test_shape_validator-1.0.0.yaml

```diff
@@ -0,0 +1,51 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://weldx.bam.de/schemas/weldx/debug/test_shape_validator-1.0.0"
+tag: "tag:weldx.bam.de:weldx/debug/test_shape_validator-1.0.0"
+
+title: |
+  simple demonstration and test schema for wx_shape validator syntax
+type: object
+properties:
+  prop1:
+    tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
+    wx_shape: [1,2,(3),(4)]
+
+  prop2:
+    tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
+    wx_shape: [3,2,1]
+
+  prop3:
+    tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
+    wx_shape: [2,4,6,8,...]
+
+  prop4:
+    tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
+    wx_shape: [1,3,5,7,9]
+
+  nested_prop:
+    type: object
+    properties:
+      p1:
+        tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
+        wx_shape: [10,8,6,4,2]
+      p2:
+        tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
+        wx_shape: [9,7,5,3,1]
+
+
+
+required: [prop1,prop2,prop3,prop4,nested_prop]
+propertyOrder: [prop1,prop2,prop3,prop4,nested_prop]
+flowStyle: block
+additionalProperties: true
+wx_shape:
+  prop1: [(~),2,n]
+  prop2: [n,2,1]
+  prop3: [2,4,5~7,...]
+  prop4: [1,3,5,k,m]
+  nested_prop:
+    p1: [10,1~10,6,4,2]
+    p2: [(m),7,5,3,1]
```

Comment:
Yes, since every instance of `wx_shape` in the yaml tree triggers it's own validation at the specific tree location.

So if you want to validate multiple shapes with matching dimensions (e.g. `n`) like
```yaml
wx_shape:
  prop1: [(~),2,n]
  prop2: [n,2,1]
```
you have to define this in a part of the tree that has all required properties in it's scope.

For example:
When defining an "inline" `wx_shape` validator with a single "variable" like
```yaml
  prop2:
    tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
    wx_shape: [3,n,1]
```
this should be equivalent to
```yaml
  prop2:
    tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
    wx_shape: [3,~,1]
```
since there is nothing else to compare the `n` to in the first example.